stream-platform/src/main/java/io/camunda/zeebe/stream/impl/StreamProcessor.java

```diff
@@ -278,6 +278,7 @@ private void tearDown() {
     streamProcessorContext.getLogStreamReader().close();
     logStream.removeRecordAvailableListener(this);
     replayStateMachine.close();
+    scheduledCommandCache.clear();
```

Comment:
💭 Since it is passed to the StreamProcessor, I'm wondering whether maybe the outside should take care of this. Like the TransitionStep.

Reply:
Hm, normally that would make sense. I didn't here because the transition step doesn't really have any state, and I assumed that was on purpose (i.e. it's always recreated and tracked by the stream processor).

We can add the cache as a field, but I did like that it was always tied to the stream proc instance to avoid cache-reuse issues :smile: